examples/scripts/dpo_online.py

```diff
@@ -69,40 +82,40 @@
     model = AutoModelForCausalLM.from_pretrained(
         model_config.model_name_or_path, trust_remote_code=model_config.trust_remote_code, **model_kwargs
     )
-    ref_model = AutoModelForCausalLM.from_pretrained(
-        model_config.model_name_or_path, trust_remote_code=model_config.trust_remote_code, **model_kwargs
-    )
+
     reward_model = AutoModelForSequenceClassification.from_pretrained(
-        training_args.reward_model_path, num_labels=1, trust_remote_code=model_config.trust_remote_code
+        training_args.reward_model_path,
+        trust_remote_code=model_config.trust_remote_code,
+        **model_kwargs,
     )
+
     tokenizer = AutoTokenizer.from_pretrained(
         model_config.model_name_or_path,
         padding_side="left",
         trust_remote_code=model_config.trust_remote_code,
+        **model_kwargs,
     )
     if tokenizer.chat_template is None:
         tokenizer.chat_template = SIMPLE_QUERY_CHAT_TEMPLATE
+    if tokenizer.pad_token_id is None:
+        tokenizer.pad_token = tokenizer.eos_token
```

Comment:
I think you need to set the ID as `pad_token` is just the string repn
```suggestion
        tokenizer.pad_token_id = tokenizer.eos_token_id
```

Reply:
Really? Because we literally have 68 occurence of this in the codebase 😅

Reply:
Ah maybe I'm wrong then. A good check would be to see if `pad_token_id` is also set to `eos_token_id` when we just set the string repns

Reply:
Both work!

```python
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("gpt2")
print(tokenizer.pad_token_id) # None
tokenizer.pad_token = tokenizer.eos_token
print(tokenizer.pad_token_id) # 50256

tokenizer = AutoTokenizer.from_pretrained("gpt2")
print(tokenizer.pad_token) # None
tokenizer.pad_token_id = tokenizer.eos_token_id
print(tokenizer.pad_token) # <|endoftext|>
```